docs/input-kafka.asciidoc

```diff
@@ -344,6 +345,27 @@ NOTE: In cases when multiple inputs are being used in a single pipeline, reading
 it's essential to set a different `group_id => ...` for each input. Setting a unique `client_id => ...`
 is also recommended.
 
+[id="plugins-{type}s-{plugin}-group_instance_id"]
+===== `group_instance_id`
+
+* Value type is <<string,string>>
+* Optional, there is no default value for this setting.
+
+The static membership identifier for this consumer. Static membership feature was introduced in
+https://cwiki.apache.org/confluence/display/KAFKA/KIP-345%3A+Introduce+static+membership+protocol+to+reduce+consumer+rebalances[KIP-345],
+available under Kafka property `group.instance.id` and aims to avoid rebalances in contexts where a lot of data
+has to be handed off after a consumer goes offline.
+This mitigates the cases where the service state is heavy and rebalance of one topic partition from instance
+A to B means huge amount of data transfer. Client that goes offline and online frequently, if using this settings
+avoid frequent and heavy rebalances.
+
+NOTE: It has to be unique across all the clients belonging to the same <<plugins-{type}s-{plugin}-group_id>>, in case another client connects
+with same `group.instance.id` value then the oldest is kicked off.
```

Comment:
```suggestion
NOTE: The `group_instance_id` setting must be unique across all the clients belonging to the same <<plugins-{type}s-{plugin}-group_id>>. Otherwise, another client connecting
with same `group.instance.id` value would cause the oldest instance to be removed.
```

Reply:
I'm not sure I got all the details right.  I'm trying to add precise language by restating the name of the option rather than just saying "it".  
Also, "kicked off" is ambiguous in English.  It can mean to start something (like kicking off a process" or deleting/removing something.  I'm guessing you mean the latter.

Reply:
Yes you guessed right :-) I thinks that instead of
```
instance to be removed
```
would be more appropriate
```
instance to be disconnected
```